x/xbasm/parse.go

```diff
@@ -0,0 +1,41 @@
+package xbasm
+
+import (
+	"errors"
+
+	"github.com/mailru/easyjson"
+
+	"github.com/blocky/basm-go-sdk"
+)
+
+type FnCallClaims struct {
+	CodeHash    string `json:"hash_of_code"`
+	Function    string `json:"function"`
+	InputHash   string `json:"hash_of_input"`
+	Output      []byte `json:"output"`
+	SecretsHash string `json:"hash_of_secrets"`
+}
+
+//easyjson:json
+type sliceOfSliceOfBytes [][]byte
+
+func ParseFnCallClaims(v basm.MarshaledAttestedObject) (*FnCallClaims, error) {
+	var fixedRep sliceOfSliceOfBytes
+	expectedCount := 5
+	switch err := easyjson.Unmarshal(v, &fixedRep); {
+	case err != nil:
+		localErr := errors.New("unmarshaling to fixed rep")
+		return nil, errors.Join(localErr, err)
+	case len(fixedRep) != expectedCount:
+		return nil, errors.New("unexpected number of fields")
+	}
+
+	c := new(FnCallClaims)
+	c.CodeHash = string(fixedRep[0])
```

Comment:
I might set up these indexes as unexported constants, but no big deal

Reply:
I'm going to leave as is.  Since these are indices that are used once and have no actual meaning, the added level of indirection would make the resulting code harder to understand.  For example, the current line would become:

``
const (
    fnCallClaimsFixedRepCodeHashIdx = 0 
    ...
)
...
// some space where other stuff happens.
...
    c.CodeHash = string(fixedRep[fnCallClaimsFixedRepCodeHashIdx])
```